Frameworks/UIKit/UITextField.mm

```diff
@@ -116,13 +122,21 @@ @implementation UITextField {
 - (void)setText:(NSString*)text {
     [_secureModeLock lock];
     if (_secureTextMode) {
-        _passwordBox.password = [text copy];
+        _passwordBox.Password(objcwinrt::string(text));
     } else {
-        _textBox.text = [text copy];
-        // Ensure caret at end of field in case we programmatically
-        // gain focus (becomeFirstResponder) after the text is set:
-        _textBox.selectionStart = [text length];
-        _textBox.selectionLength = 0;
+        _textBox.Text(objcwinrt::string(text));
+
+        try {
+            // Ensure caret at end of field in case we programmatically
+            // gain focus (becomeFirstResponder) after the text is set:
+            _textBox.SelectionStart([text length]);
+            _textBox.SelectionLength(0);
+        } catch (const winrt::hresult_error& e) {
+            // Bug in XAML TextBox causes it to return S_FALSE
```

Comment:
Actually it looks like the C++/WinRT headers from NuGet have this fix, so I've removed the try/catch entirely.

---
In reply to: [109244741](https://github.com/Microsoft/WinObjC/pull/2382#discussion_r109244741) [](ancestors = 109244741,109038960)